### PR TITLE
Add photoelectric process and Livermore photoelectric model/interact kernel

### PR DESCRIPTION
**Pull request by amandalund**
This adds a `PhotoelectricProcess` and `LivermorePEModel`, plus the interaction kernel for the model. I added a basic test for the model, which I can expand if we want (e.g., do we want to test the `interact` method?)

### CODE REVIEW

**Review by sethrj — CHANGES_REQUESTED** [review]

File: src/physics/em/detail/LivermorePE.cu [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        return;
+
+    SecondaryAllocatorView allocate_secondaries(ptrs.secondaries);
+    ParticleTrackView particle(ptrs.params.particle, ptrs.states.particle, tid);
+    MaterialTrackView material(ptrs.params.material, ptrs.states.material, tid);
+    PhysicsTrackView  physics(ptrs.params.physics,
+                             ptrs.states.physics,
+                             particle.def_id(),
+                             material.def_id(),
+                             tid);
```
[the remark this answers is not in the document]
Comment (amandalund) [reply src/physics/em/detail/LivermorePE.cu:47]: Weird, this actually was the result of running clang-format.. I'm not sure why only the top line got shifted.

[the remark this answers is not in the document]
Comment (sethrj) [reply src/physics/em/detail/LivermorePE.cu:47]: Oh, ok. Well in that case it's just clang being weird. 🙄 Thanks though!

File: test/physics/em/LivermorePE.test.cc [hunk 2]
```diff
@@ -238,3 +236,26 @@ TEST_F(PhotoelectricInteractorTest, stress_test)
         = {15.99755859375, 16.09204101562, 13.79919433594, 8.590209960938, 2};
     EXPECT_VEC_SOFT_EQ(expected_avg_engine_samples, avg_engine_samples);
 }
+
+TEST_F(LivermorePEInteractorTest, model)
+{
+    PhotoelectricProcess process(particle_params_, livermore_params_);
+    ModelIdGenerator     next_id;
```
[the remark this answers is not in the document]
Comment (amandalund) [reply test/physics/em/LivermorePE.test.cc:243]: Sounds good, maybe we can discuss at the meeting today how we want to restructure things. Should I start tackling that here?

[the remark this answers is not in the document]
Comment (sethrj) [reply test/physics/em/LivermorePE.test.cc:243]: Yeah, let's chat about it at 2. I don't think you need to tackle it here though -- I was just commenting as a "todo" for later.

**Review by sethrj — APPROVED** [review]

**Comment by sethrj** [comment]
Ver nice, thanks @amandalund !

**Pull request merged**